test/box-luatest/lua_memory_test.lua

```diff
@@ -0,0 +1,68 @@
+local server = require('luatest.server')
+local t = require('luatest')
+local g = t.group('lua_memory setting test')
+--local treegen = require('test.treegen')
+
+local MB = 1024 * 1024
+local LUA_MIN_MEMORY = 256 * MB
+
+g.before_each(function()
+    g.server = server:new{
+        alias = 'default',
+        box_cfg = {lua_memory = LUA_MIN_MEMORY}
+    }
+    g.server:start()
+
+end)
+
+g.after_each(function()
+    g.server:drop()
+end)
+
+g.test_lua_memory_setting = function(g)
+    g.server:exec(function()
+        local MB = 1024 * 1024
+        local LUA_MIN_MEMORY = 256 * MB
+        local LUA_EXTENDED_MEMORY = 600 * MB
+
+        t.assert_equals(box.cfg.lua_memory, LUA_MIN_MEMORY)
+
+        -- Increasing the limit should be ok.
+        box.cfg{lua_memory = LUA_MIN_MEMORY + 1}
+        t.assert_equals(box.cfg.lua_memory, LUA_MIN_MEMORY + 1)
+
+        local err_min = 'Cannot limit the Lua memory with values less than '..
+                        '256MB'
+        -- Setting the limit less than 256MB is prohibited.
+        t.assert_error_msg_equals(err_min, box.cfg,
+            {lua_memory = LUA_MIN_MEMORY - 1})
+
+        box.cfg{lua_memory = LUA_EXTENDED_MEMORY}
+
+        local used = collectgarbage('count') * 1024
+        local free = LUA_EXTENDED_MEMORY - used
+        local _ = string.rep('a', free / 3)
+
+        local err_decreasing = 'Cannot limit the Lua memory with values less '..
+                               'than the currently allocated amount'
+        -- Decreasing less than currently allocated is forbidden.
+        t.assert_error_msg_equals(err_decreasing, box.cfg,
+            {lua_memory = LUA_MIN_MEMORY})
+
+    end)
+
+end
+
+g.test_lua_memory_semantics = function()
+    g.server:exec(function()
+        local memory = 256 * 1024 * 1024
+
+        local function overflow()
+            local free = memory - (collectgarbage('count')) * 1024
+            local _ = string.rep('a', free / 2)
+        end
+
+        local err_min = 'not enough memory'
+        t.assert_error_msg_equals(err_min, overflow)
+    end)
```

Comment:
May be it is worth to check the `box.error.last()` too.

Reply:
Done!

Reply:
```diff
+        local err1 = 'not enough memory'
+        local err2 = 'Failed to allocate'
+        t.assert_error_msg_equals(err1, overflow)
+        t.assert_str_contains(tostring(box.error.last()), err2)
```

I suppose we can just drop local variables (`err1`, `err2`) here for better readability.

```lua
        t.assert_error_msg_equals('not enough memory', overflow)
        t.assert_str_contains(tostring(box.error.last()), 'Failed to allocate')
```
What do you think?

Reply:
Yes, you're right. The strings are short enough. Done!